perf/lua/column_insert_module.c

```diff
@@ -151,46 +211,87 @@ arrow_array_init(struct ArrowArray *array, int row_count)
 			.release = arrow_array_destroy,
 			.private_data = NULL,
 		};
+		int num = column_numbers[i];
+		assert(num < column_count);
+		array->children[i]->buffers[1] =
+			&dataset.columns[num].data[row_offset];
```

Comment:
Don't we still insert first column_count columns after that?

We still have `column_count` childs in the array and I see no place we specify column numbers the children represent.

I don't quite familiar with the arrow API, so the question may be too obvious to ask.

Reply:
No, columns are specified by names in `arrow_schema_init()`.